ROADMAP.md

```diff
@@ -1,31 +1,41 @@
 # Roadmap
 
-This document defines a high level roadmap for Rook development. The dates below are subject to change but should give a general idea of what we are planning. We use the [milestone](https://github.com/rook/rook/milestones) feature in Github so look there for the most up-to-date and issue plan.
+This document defines a high level roadmap for Rook development.
+The dates below are subject to change but should give a general idea of what we are planning.
+We use the [milestone](https://github.com/rook/rook/milestones) feature in Github so look there for the most up-to-date and issue plan.
 
-## Rook 0.7
+## Rook 0.8
 
-- Ceph Block Storage (Cluster and Pool CRDs) declared beta
-- Mon reliability (restarts, failing over too fast, ip changes, etc.)
-- Durability of state (local storage support, config is regenerat-able)
-- Run everywhere (rook-agent uses ceph-fuse, nbd-rbd / tcmu runner)
-- Adding / removing nodes and disk drives (lifecycle issues, failures, etc.)
-- Run with Least Privileged and possibly without privileged containers
-- Shutdown / restart issues
-- Improved data placement and pool configuration (CRUSH maps)
-- Placement group balancer support (ceph-mgr module)
-- Use new Prometheus ceph-mgr module
-- Performance Testing for Block storage
-- Make API and CLI Optional (plan to remove)
-- Dynamic Volume Provisioning for CephFS
+* Multiple storage backends
+  * Full design
+  * Refactor code base and repositories to enable
+  * Consider support for Minio, potentially early support for other backends time permitting
+* Switch CRDs to API Aggregation
+* Run on arbitrary PVs
+* Remove Rook API and CLI
+* Migrate CI and release pipelines to a solution hosted by the CNCF
+* Run with Least Privileged and possibly without privileged containers
+* Shutdown / restart issues
+* Support Kubernetes 1.7+ only
+* Ceph features and improvements
+  * Disk management (1 OSD per pod, adding/removing disks)
+  * Placement group balancer support (ceph-mgr module)
+  * Mon reliability (restarts, failing over too fast, ip changes, etc.)
+  * Mimic support
 
```

Comment:
Please consider another feature to run ceph daemon in separate container image. Right now, rook and ceph daemons (mon, osd, mgr) colocate in the same container image. This is a problem when rook and ceph daemons are not required to upgrade at the same time. Separating rook from ceph daemons makes container images upgrade easy to handle

Reply:
agreed, this would fall under the disk management features. We may not need to call it out at this level of the road map, or maybe we add another comment in parenthesis: `, separate provisioning from the daemon`

Reply:
i think @rootfs is asking for something more broad of scope than separating the disk provisioning from the running of the daemon.  it's the general question of what image is being used to run the ceph daemons? if it has rook code in it, then it has to get updated when rook version increments, regardless of if there is a ceph release update as well.  i think #1404 captures the discussion around this, which is very much part of the multiple backend design and discussion, already included as an item on the road map.